hack/local-up-master/master.sh

```diff
@@ -0,0 +1,316 @@
+    KUBE_APISERVER_PID=$!
+
+    # Wait for kube-apiserver to come up before launching the rest of the components.
+    echo "Waiting for kube-apiserver to come up"
+    kube::util::wait_for_url "https://${API_HOST_IP}:${API_SECURE_PORT}/healthz" "kube-apiserver: " 1 ${WAIT_FOR_URL_API_SERVER} ${MAX_TIME_FOR_URL_API_SERVER} \
+        || { echo "check kube-apiserver logs: ${KUBE_APISERVER_LOG}" ; exit 1 ; }
+
+    # Create kubeconfigs for all components, using client certs
+    kube::util::write_client_kubeconfig "${CONTROLPLANE_SUDO}" "${CERT_DIR}" "${ROOT_CA_FILE}" "${API_HOST}" "${API_SECURE_PORT}" admin
+    chown "${USER}" "${CERT_DIR}/client-admin.key" # make readable for kubectl
+}
+
+function start_kubecontrollermanager {
+    if [ ! -d "${CONFIG}/kube-controller-manager" ]; then
+        mkdir -p ${CONFIG}/kube-controller-manager
+        generate_kubecontrollermanager_certs
+    fi
+
+    KUBE_CONTROLLER_MANAGER_LOG=${LOG_DIR}/kube-controller-manager.log
+    hyperkube controller-manager \
+      --v=${LOG_LEVEL} \
+      --vmodule="${LOG_SPEC}" \
+      --service-account-private-key-file="${CONFIG}/kube-controller-manager/etcd-serving-ca.crt" \
+      --root-ca-file="${ROOT_CA_FILE}" \
+      --kubeconfig  ${CONFIG}/kube-controller-manager/controller.kubeconfig \
+      --use-service-account-credentials \
+      --leader-elect=false >"${KUBE_CONTROLLER_MANAGER_LOG}" 2>&1 &
+    KUBE_CONTROLLER_MANAGER_PID=$!
+
+    echo "Waiting for kube-controller-manager to come up"
+    kube::util::wait_for_url "http://localhost:10252/healthz" "kube-controller-manager: " 1 ${WAIT_FOR_URL_API_SERVER} ${MAX_TIME_FOR_URL_API_SERVER} \
+        || { echo "check kube-controller-manager logs: ${KUBE_CONTROLLER_MANAGER_LOG}" ; exit 1 ; }
+}
+
+function start_openshiftapiserver {
+    if [ ! -d "${CONFIG}/openshift-apiserver" ]; then
+        mkdir -p ${CONFIG}/openshift-apiserver
+        cp ${OS_ROOT}/hack/local-up-master/openshift-apiserver.yaml ${CONFIG}/openshift-apiserver
+        generate_openshiftapiserver_certs
+    fi
+
+    OPENSHIFT_APISERVER_LOG=${LOG_DIR}/openshift-apiserver.log
+    hypershift openshift-apiserver \
+      --v=${LOG_LEVEL} \
+      --vmodule="${LOG_SPEC}" \
+      --config=${CONFIG}/openshift-apiserver/openshift-apiserver.yaml >"${OPENSHIFT_APISERVER_LOG}" 2>&1 &
+    OPENSHIFT_APISERVER_PID=$!
+
+    # Wait for openshift-apiserver to come up before launching the rest of the components.
+    echo "Waiting for openshift-apiserver to come up"
+    kube::util::wait_for_url "https://${API_HOST_IP}:8444/healthz" "openshift-apiserver: " 1 ${WAIT_FOR_URL_API_SERVER} ${MAX_TIME_FOR_URL_API_SERVER} \
+        || { echo "check kube-apiserver logs: ${OPENSHIFT_APISERVER_LOG}" ; exit 1 ; }
+
+    NON_LOOPBACK_IPV4=$(ifconfig | grep -A1 -e "wlp\|enp" | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*' | grep -v '127.0.0.1' | head -n 1)
```

Comment:
`ip -o -4 addr show up primary scope global dynamic | awk '{print $4}' | cut -f1  -d'/'` is shorter and much more readable imo.